File: Data Structures & Algorithms/two-integer-sum/submission-0.go
func twoSum(nums []int, target int) []int {
    valueIdxMap := make(map[int]int)
	for i, num := range nums {
		valueIdxMap[num] = i
	}
	for i, num := range nums {
		diff := target - num
		if j, found := valueIdxMap[diff]; found && j != i {
			return []int{i, j}
		}
	}
	return []int{}
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1
        while l < r:
            curr_sum = numbers[l] + numbers[r]
            if curr_sum == target:
                return [l+1, r+1]
            if curr_sum < target:
                l += 1
            else:
                r -= 1
        return [l+1, r+1]
        